openrtb_ext/request.go

```diff
@@ -151,14 +151,15 @@ type ExtRequestPrebidCacheVAST struct {
 
 // ExtRequestTargeting defines the contract for bidrequest.ext.prebid.targeting
 type ExtRequestTargeting struct {
-	PriceGranularity     *PriceGranularity        `json:"pricegranularity,omitempty"`
-	IncludeWinners       *bool                    `json:"includewinners,omitempty"`
-	IncludeBidderKeys    *bool                    `json:"includebidderkeys,omitempty"`
-	IncludeBrandCategory *ExtIncludeBrandCategory `json:"includebrandcategory,omitempty"`
-	IncludeFormat        bool                     `json:"includeformat,omitempty"`
-	DurationRangeSec     []int                    `json:"durationrangesec,omitempty"`
-	PreferDeals          bool                     `json:"preferdeals,omitempty"`
-	AppendBidderNames    bool                     `json:"appendbiddernames,omitempty"`
+	PriceGranularity          *PriceGranularity          `json:"pricegranularity,omitempty"`
+	MediaTypePriceGranularity *MediaTypePriceGranularity `json:"mediatypepricegranularity,omitempty"`
```

Comment:
Wow! This is a great effort!
Provided example is a good way to demonstrate the difference, however it's far from what happens in actual code including data creation. In your example you create data directly like 
```go
return &FooWithBarPtr{
		bar: &Bar{x: &x, y: &y},
	}
}
```
in real prod code all objects are created by json unmarshal. 

[Here](https://go.dev/play/p/ErFj4m9ZB_6) is the test with similar approach, but a bit closer to what we have in prod.

Here is the result:

```
go test -bench . -benchmem
goos: darwin
goarch: amd64
pkg: github.com/prebid/prebid-server
cpu: Intel(R) Core(TM) i9-9880H CPU @ 2.30GHz
BenchmarkTargetingPtrNoData-16                   2089046               566.0 ns/op           248 B/op          7 allocs/op
BenchmarkTargetingNoPtrNoData-16                 2011388               597.1 ns/op           264 B/op          7 allocs/op
BenchmarkTargetingPtrWithData-16                  176634              6913 ns/op             944 B/op         31 allocs/op
BenchmarkTargetingNoPtrWithData-16                179672              6633 ns/op             936 B/op         30 allocs/op
BenchmarkTargetingPtrNoDataModify-16             2089758               555.4 ns/op           248 B/op          7 allocs/op
BenchmarkTargetingNoPtrNoDataModify-16           2097586               577.9 ns/op           264 B/op          7 allocs/op
BenchmarkTargetingPtrWithDataModify-16            177097              6546 ns/op             944 B/op         31 allocs/op
BenchmarkTargetingNoPtrWithDataModify-16          177206              6833 ns/op             936 B/op         30 allocs/op
PASS

```
I consider them equal. Run it couple of times to see how it varies.

I went through [Benchmark results](https://pkg.go.dev/testing#BenchmarkResult) explanation and still don't understand how this shows GC usage (improvement or deterioration), please explain, it will be useful for future.

Reply:
Discussed offline with the team, decided to change to no-pointer.